### PR TITLE
#1886 - LPAR/PPC64 bootlist is incorrectly set when having multiple 'prep' partitions

### PR DESCRIPTION
**Pull request by rmetrich**
#### Relax-and-Recover (ReaR) Pull Request Template

Please fill in the following items before submitting a new pull request:

##### Pull Request Details:

* Type: **Bug Fix**

* Impact: **Normal**

* Reference to related issue (URL): [https://github.com/rear/rear/issues/1886](#1886)

* How was this pull request tested?

    Tested on AIX/LPAR/PPC64le with 2 prep partitions, but no multipath on older ReaR 2.00 version

* Brief description of the changes in this pull request:

    - Added handling of multiple *prep* partitions
    - Enhanced handling of multiple *prep* partitions and multipath

### CODE REVIEW

File: usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh [hunk 1]
```diff
@@ -29,14 +32,23 @@ if [ -n "$part" ]; then
...
+
+if [[ ${#boot_list[@]} -gt 5 ]]; then
+    LogPrint "Too many entries for bootlist command, limiting to first 5 entries..."
+    boot_list=( ${boot_list[@]:0:5} )
+fi
+
+if [[ ${#boot_list[@]} -gt 0 ]]; then
+    LogPrint "Set LPAR bootlist to '${boot_list[@]}'"
+    bootlist -m normal $boot_list
     LogIfError "Unable to set bootlist. You will have to start in SMS to set it up manually."
```
Comment (jsmeix) [inline usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh]: Regadless that this is not part of this pull request but in general
I think `LogIfError` is useless to tell something to the user
because the `LogIfError` output appears only in the log file.
To show an error messare to the user better use `LogPrintIfError`
or whatever appropriate function in lib/_input-output-functions.sh

**Comment by jsmeix** [comment]
@schabrolles 
could you have a look here an review it?
If you don't have time I would "just merge" it tomorrow
because it was tested by @rmetrich

**Comment by jsmeix** [comment]
@schabrolles 
I would like to "just merge" it today afternoon if you do not object, cf.
https://github.com/rear/rear/pull/1887#issuecomment-410659457

**Comment by schabrolles** [comment]
@jsmeix, bootlist is responsible to set the boot devices order list for an LPAR in PowerVM.
Unfortunately, My PowerVM system is used for a customer test until end of August. So I'm not able to perform a test to validate it.

**Comment by schabrolles** [comment]
@jsmeix, for a plain looking code, this looks correct to me...

**Comment by jsmeix** [comment]
@rmetrich 
many thanks for your fix and enhancements!

@schabrolles 
many thanks for your review!

File: usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh [hunk 2]
```diff
@@ -29,14 +32,23 @@ if [ -n "$part" ]; then
...
+done
+
+if [[ ${#boot_list[@]} -gt 5 ]]; then
+    LogPrint "Too many entries for bootlist command, limiting to first 5 entries..."
+    boot_list=( ${boot_list[@]:0:5} )
 fi
+
+if [[ ${#boot_list[@]} -gt 0 ]]; then
+    LogPrint "Set LPAR bootlist to '${boot_list[@]}'"
+    bootlist -m normal $boot_list
```
Comment (pcahyna) [inline usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh]: Expanding $boot_list this way will return only the first element. Is it intended? Why is boot_list an array if it is intended - one element would be enough?
IIUC, this will lead to boot failure if the first path is disconnected.

Comment (rmetrich) [reply usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh]: Ough, this is indeed broken!

File: usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh [hunk 3]
```diff
@@ -29,14 +32,23 @@ if [ -n "$part" ]; then
...
-    LogIfError "Unable to set bootlist. You will have to start in SMS to set it up manually."
+done
+
+if [[ ${#boot_list[@]} -gt 5 ]]; then
+    LogPrint "Too many entries for bootlist command, limiting to first 5 entries..."
+    boot_list=( ${boot_list[@]:0:5} )
 fi
+
+if [[ ${#boot_list[@]} -gt 0 ]]; then
+    LogPrint "Set LPAR bootlist to '${boot_list[@]}'"
```
Comment (pcahyna) [inline usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh]: Using `${boot_list[@]}` inside double quotes will cause it to expand to multiple strings, like this:
```
boot_list=( foo bar )
"Set LPAR bootlist to '${boot_list[@]}'" -> "Set LPAR bootlist to 'foo"   "bar'"
```
LogPrint actually handles multiple arguments properly, so it is not a bug. But it is IMO better to do this in a less surprising way, because I don't think this kind of splitting into multiple argument was intended. And it can become a bug when this code gets cargo-culted to other places. And it is flagged by static analysis: https://github.com/koalaman/shellcheck/wiki/SC2145.
I propose:
```
LogPrint "Set LPAR bootlist to '${boot_list[*]}'"
```
(mentioned also in #2097 )

Comment (pcahyna) [reply usr/share/rear/finalize/Linux-ppc64le/680_install_PPC_bootlist.sh]: I filed #2098 for this.
